ui/perfherder/graphs/GraphsView.jsx

```diff
@@ -55,6 +55,17 @@ class GraphsView extends React.Component {
     this.checkQueryParams();
   }
 
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.location.search === '' &&
+      this.state.testData.length !== 0 &&
+      this.state.loading !== true &&
+      this.props.location.search !== prevProps.location.search
+    ) {
+      this.resetPageOnHeaderClick();
```

Comment:
We don't need to hide the setstate, we should use it directly and suppress the lint warning

```
    {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        testData: [],
      });
    }
```